### ISSUE TITLE
How to solve loss is NAN?

### ISSUE THREAD

**Issue by Jhy1993** [xiangwang1223/neural_graph_collaborative_filtering#17]
Hi
Thanks for your brilliant code. However, I run your code several times and always meet "loss is NAN".
So how can I avoid this phenomenon?

**Comment by xiangwang1223** [comment]
Hi,
Thanks for your interest. According to my experience, please try to make the learning rate and l2_normalization smaller, say 10e-5. Thanks.

**Comment by xiangwang1223** [comment]
Hi,
I learn from my peers. Maybe you can try the following code to modify the BPR loss, in order to avoid NAN:

loss = tf.reduce_sum(tf.nn.softplus(-(pos_result - neg_result)))

Please let me know whether it works. Thanks.

**Comment by Jhy1993** [comment]
It works！ 
Thanks for your advice.
